docs/specs/lcb-packages.md

```diff
@@ -0,0 +1,192 @@
+
+### Mimetype (pending)
+
+This file has a single line of text which gives the MIME type for the document.
+It must be the first file included in the zip archive and be stored in an
+uncompressed form. This ensures that 'magic number' mechanisms can be used to
+determine the actual type of the package file.
+
+### Manifest (current)
+The manifest.xml file is generated by the compiler when it compiles a module.
+This file replicates most of the information present in the binary module file,
+but in a way the loading environment can access directly without having to load
+the module.
+
+The outline of the manifest file format is as follows:
+
+    <package version="1.0">
+        <label>Human Readable Foo</label>
+        <author>Mr Magoo</author>
+        <description>Foo is a super amazing widget that will do everything for you.</description>
+        <license>commercial|dual|community</license>
+        <name>com.livecode.foo</name>
+        <version>X.Y.Z</version>
+        <type>widget|library</type>
+        <requires name="com.livecode.bar" version="X.Y.Z" />
+        <requires name="com.livecode.baz" version="X.Y.Z" />
+        <property name="foo" get="optional(integer)" set="optional(integer)" />
+        <property name="bar" get="string" />
+        <event name="click" parameters="in(integer),out(real)" return="optional(any)" />
+        <handler name="magic" parameters="in(integer),inout(string)" return="undefined" />
+    </package>
+
+This format works where a package contains a single module, and that module is a
+library or widget. However, it is not sufficient to describe a package which
+exports multiple widgets or library exports.
+
+### Manifest (revised)
+
+TBD
+
+### Modules (revised)
+
+The modules folder contains all the compiled module code which the package contains.
+The filenames of each module file must match the name of module contained within it
+with a .lcb suffix.
+
+A package is considered invalid if the package name is not a common strict prefix of
+all the modules it contains.
+
+For example, this is a valid package:
+
+    com.example.foo.lcp/
+        modules/
+            com.example.foo.bar.lcb
+            com.example.foo.bar.tests.lcb
+            
+This is an invalid package:
+
+    com.example.foo.lcp/
+        modules/
+            com.example.foo.lcb
+            org.example.foo.lcb
+
+### Source
+
+The source folder contains the original source files which were used to build the
+compiled modules contained within the modules folder. The original folder structure
+of the source will be preserved as much as possible, with the debug annotations
+within the compiled modules themselves referencing files relative to <package>/source.
+
+### Code (revised)
+
+The code folder contains all the (non-system) foreign code components that are
+needed by any compiled module 'foreign handler' definitions.
+
+The folder is split up into sub-folders by architecture triples:
+
+    <platform>-<arch>-<variant>

```

Comment:
I'd prefer the scheme to be:

<platform>/<arch>/<variant>

(this is what we use for our pre-built libraries). It also handles universal libraries properly (by omitting the arch):

osx/MyFooLib-Univeral.dylib
win32/x86/MyFooLib.dll
linux/x86/MyFooLib.so
ios/armv7/iPhoneOS6.1/MyFooLib.lib
ios/iPhoneOS8.1/MyFooLib.lib
ios/x86/iPhoneSimulator6.1/MyFooLib.dylib
ios/iPhoneSimulator8.1/MyFooLib.dylib
android/java/MyFooLib.jar